test/mpi/Makefile.mtest

```diff
@@ -19,16 +19,21 @@ AM_CPPFLAGS = -I$(top_builddir)/include -I$(top_srcdir)/include
 LDADD = $(top_builddir)/util/mtest.$(OBJEXT) $(top_builddir)/util/mtest_common.$(OBJEXT)
 
 # Add libdtpools support
+DTPOBJS = $(top_builddir)/dtpools/src/.libs/libdtpools_la-dtpools.o             \
+          $(top_builddir)/dtpools/src/.libs/libdtpools_la-dtpools_attr.o        \
+          $(top_builddir)/dtpools/src/.libs/libdtpools_la-dtpools_desc.o        \
+          $(top_builddir)/dtpools/src/.libs/libdtpools_la-dtpools_init_verify.o \
+          $(top_builddir)/dtpools/src/.libs/libdtpools_la-dtpools_misc.o
```

Comment:
I don't think we should be relying on `.libs` content.  That's internal libtools stuff, and the naming can change from version to version.  I think the autotools-intended way to do this is to make dtpools a convenience library and have the tests depend on it.

Reply:
I know and I agree with you this is not an ideal solution. I remember trying to link dtpools as external library in the first dtpools implementation but it didn't work. I will try again and see if I can make that work.

Reply:
`.libs` is for objects built with `PIC` that is for shared library. I believe the static objects are directly in the `src` directory. Therefore, when it is configured with static only, those `.libs` objects won't be available.

Per [libtool manual](https://www.gnu.org/software/libtool/manual/html_node/Linking-executables.html#Linking-executables), Directly linking with `libdtpools.la` will create executable in `.libs` and the executable become a shell script wrapper. It is supposed to work (as runnable), but each will incur overhead (nearly half second according to the example). I think for testing purpose, we probably should prefer testing the binary directly. Then we need install dtpools before running the tests somehow.

Reply:
A compromise solution maybe to intercept the configure options to `test/mpi` and always configure `dtpools` `static-only`. That way, we can leave `dtpools` intact, i.e. still fully supports static or dynamic build if someone choose to use `dtpools` independently. We also simplify the testsuite build -- always have dtpools statically linked in. Anyway, when user configure mpich with `--disable-static`, they really only meant that option for `mpich`, not testsuite. So the solution of isolating the option of `testsuite` from that of `mpich` better reflects user's intention.

Reply:
Always building dtpools statically for internal use sounds like a reasonable compromise to me.

Reply:
> I remember trying to link dtpools as external library in the first dtpools implementation but it didn't work. 

Chatting with Ken and realize the problem is obvious: to compile tests without installing `dtpool` only works if we build tests with `libtool` wrapper. Since we are using `mpicc` etc to build tests, that won't work.

Reply:
> A compromise solution maybe to intercept the configure options to `test/mpi` and always configure `dtpools` `static-only`. That way, we can leave `dtpools` intact, i.e. still fully supports static or dynamic build if someone choose to use `dtpools` independently. We also simplify the testsuite build -- always have dtpools statically linked in. Anyway, when user configure mpich with `--disable-static`, they really only meant that option for `mpich`, not testsuite. So the solution of isolating the option of `testsuite` from that of `mpich` better reflects user's intention, as well as fits our future direction (of making test suite independent)

I just saw this comment, yes that might be an option. I think Ken was also suggesting a similar approach.

Reply:
@pavanbalaji I think this is closer to what I had implemented in patch #2 except we don't want to expose dtpools externally